src/main/webapp/app/shared/profile-toggle/profile-toggle.directive.ts

```diff
@@ -0,0 +1,68 @@
+import { Directive, HostBinding, Input, OnDestroy, OnInit } from '@angular/core';
+import { tap } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+import { ProfileToggle, ProfileToggleService } from 'app/shared/profile-toggle/profile-toggle.service';
+
+@Directive({
+    selector: '[jhiProfileToggle]',
+})
+export class ProfileToggleDirective implements OnInit, OnDestroy {
+    @Input('jhiProfileToggle') profiles: ProfileToggle | ProfileToggle[];
+    /**
+     * This input must be used to overwrite the disabled state given that the profile toggle is inactive.
+     * If the normal [disabled] directive of Angular would be used, the HostBinding in this directive would always enable the element if the profile is active.
+     */
+    @Input() overwriteDisabled: boolean | null;
+    /**
+     * Condition to check even before checking for the profile toggle. If true, the profile toggle won't get checked.
+     * This can be useful e.g. if you use the same button for different profiles (like our delete button) and only want
+     * to check the toggle for programming exercises
+     */
+    @Input() skipProfileToggle: boolean;
+    private profileActive = true;
+
+    private profileToggleActiveSubscription: Subscription;
+
+    constructor(private profileToggleService: ProfileToggleService) {}
+
+    /**
+     * Life cycle hook called by Angular to indicate that Angular is done creating the component
+     */
+    ngOnInit() {
+        // If no profile is set for the toggle, the directive does nothing.
+        if (!this.profiles) {
+            return;
+        }
```

Comment:
question: Can `this.profiles` be `undefined`?

Reply:
It could be undefined if the profile is added conditionally (e.g. when a component is reused for Quiz and Modelling exercises, it may receive "ProfileToggle.QUIZ" as input for quizzes but undefined for modelling exercises (because they are not yet part of their own subsystem).